src/unitxt/artifact.py

```diff
@@ -456,20 +509,27 @@ def fetch_artifact(artifact_rep) -> Tuple[Artifact, Union[AbstractCatalog, None]
     (5) Otherwise, check that the artifact representation is a dictionary and build an Artifact object from it.
     """
     if isinstance(artifact_rep, Artifact):
+        if isinstance(artifact_rep, ArtifactLink):
+            return artifact_rep.artifact_linked_to, None
         return artifact_rep, None
 
     # If local file
     if isinstance(artifact_rep, str) and Artifact.is_artifact_file(artifact_rep):
-        return Artifact.load(artifact_rep), None
+        artifact_to_return = Artifact.load(artifact_rep)
+        if isinstance(artifact_rep, ArtifactLink):
+            artifact_to_return = artifact_to_return.artifact_linked_to
```

Comment:
Same here.  The fetch_artificat should return the artificat, not the name of the artifact we link to.

Reply:
Hi @yoavkatz , and thanks again for your neat idea about the dedicated `load` of `ArtifactLink`. I just expanded it to also allow link to link to link (imagine that..  with the `overwrites` that @elronbandel emphasized floating all the way over the links in the middle, until they can be applied on the final, non-link `Artifact` that can digest them - only it is expected to have these keys that show in the overwrites).

And thanks for hour new comment. I see what you mean, and now fixing.

Reply:
Hi @yoavkatz , I also added the requested documentation:
![image](https://github.com/user-attachments/assets/60841ef2-5b76-4a87-9c1a-56939f9ab24e)

Reply:
Thanks!   I've added some followup feedback.